lib/system/linux/device.c

```diff
@@ -234,7 +234,7 @@ static int metal_uio_dev_open(struct linux_bus *lbus, struct linux_device *ldev)
 		result = (result ? result :
 			 metal_uio_read_map_attr(ldev, i, "size", &size));
 		result = (result ? result :
-			 metal_map(ldev->fd, offset, size, 0, 0, &virt));
+			 metal_map(ldev->fd, i * getpagesize(), size, 0, 0, &virt));
```

Comment:
`offset` (line 152) variable is not used any more, should I remove it? It may failed to build if -Werror and -Wunused compilation flag are turned on together.